internal/bucket/lifecycle/lifecycle.go

```diff
@@ -171,15 +172,18 @@ func (lc Lifecycle) HasActiveRules(prefix string) bool {
 		if !rule.NoncurrentVersionTransition.IsNull() {
 			return true
 		}
-		if rule.Expiration.IsNull() && rule.Transition.IsNull() {
-			continue
-		}
 		if !rule.Expiration.IsDateNull() && rule.Expiration.Date.Before(time.Now().UTC()) {
 			return true
 		}
 		if !rule.Expiration.IsDaysNull() {
 			return true
 		}
+		if rule.Expiration.DeleteMarker.val {
+			return true
+		}
+		if rule.Expiration.DeleteAll.val {
```

Comment:
Found that `ExpiredObjectAllVersions` extension to was affected by this. I have added a unit test to capture that as well.

Reply:
Update: I was wrong about this. `ExpiredObjectAllVersions` requires that `Days` is greater than zero. This rule type isn't affected by the bug this PR is addressing.